### PR TITLE
make DNS parsing more robust

### LINKED ISSUE #74: async-dns parser throws on unexpected record type
I'm seeing the following exception with reactive-lib 0.9.0 under GKE, when Akka Cluster Bootstrap performs discovery:

```
14:49:38 ERROR OneForOneStrategy [akka://iris/system/IO-DNS/async-dns/$a/$a] - key not found: 880
java.util.NoSuchElementException: key not found: 880
        at scala.collection.MapLike.default(MapLike.scala:232)
        at scala.collection.MapLike.default$(MapLike.scala:231)
        at scala.collection.AbstractMap.default(Map.scala:59)
        at scala.collection.mutable.HashMap.apply(HashMap.scala:65)
        at scala.Enumeration.apply(Enumeration.scala:137)
        at com.lightbend.rp.asyncdns.raw.RecordType$.parse(RecordType.scala:35)
        at com.lightbend.rp.asyncdns.raw.Question$.parse(Question.scala:20)
        at com.lightbend.rp.asyncdns.raw.Message$.$anonfun$parse$1(Message.scala:109)
        at com.lightbend.rp.asyncdns.raw.Message$.$anonfun$parse$1$adapted(Message.scala:109)
        at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
        at scala.collection.immutable.Range.foreach(Range.scala:156)
        at scala.collection.TraversableLike.map(TraversableLike.scala:234)
        at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at com.lightbend.rp.asyncdns.raw.Message$.parse(Message.scala:109)
        at com.lightbend.rp.asyncdns.raw.DnsClient$$anonfun$ready$1.applyOrElse(DnsClient.scala:65)
        at akka.actor.Actor.aroundReceive(Actor.scala:517)
        at akka.actor.Actor.aroundReceive$(Actor.scala:515)
        at com.lightbend.rp.asyncdns.raw.DnsClient.aroundReceive(DnsClient.scala:12)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)
        at akka.actor.ActorCell.invoke(ActorCell.scala)
...
```

I'm not sure if 880 actually corresponds to a returned record type or if the parsing is otherwise confused.

The exception breaks the cluster bootstrap process.

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes #74

### CODE REVIEW

**Comment by eed3si9n** [comment]
I am not sure where 880 is coming from either.
It's known that there are some non-standard records that provide CNAME for apex records (`ALIAS`, [`ANAME`](https://tools.ietf.org/html/draft-ietf-dnsop-aname-01)), but not sure what the number would be.

File: akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/ClusterStatusCheck.scala [hunk 1]
```diff
@@ -24,24 +24,24 @@ import scala.concurrent.{ ExecutionContext, Future }
...
-      case MemberStatus.Leaving  => false
-      case MemberStatus.Exiting  => false
-      case MemberStatus.Down     => false
-      case MemberStatus.Removed  => false
+      case MemberStatus.Up => true
+      case MemberStatus.Leaving => false
+      case MemberStatus.Exiting => false
+      case MemberStatus.Down => false
+      case MemberStatus.Removed => false
     }
```
Comment (ktoso) [inline akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/ClusterStatusCheck.scala]: try not to break formatting please, makes it harder to track changes

Comment (eed3si9n) [reply akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/ClusterStatusCheck.scala]: I am committing what sbt-scalariform did. Not sure how @dwijnand got this to compile in 72e67d236bf6df49851f47ef7352002335ce7439.

Comment (dwijnand) [reply akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/ClusterStatusCheck.scala]: Formatting isn't enforced by CI.

**Pull request merged**